src/zcl_aff_log.clas.testclasses.abap

```diff
@@ -0,0 +1,213 @@
+      two_messages_for_one_object FOR TESTING RAISING cx_static_check,
+      add_catched_exception FOR TESTING RAISING cx_static_check,
+      add_classic_exception FOR TESTING RAISING cx_static_check,
+      get_sy_message FOR TESTING RAISING cx_static_check,
+      add_exception_as_info FOR TESTING RAISING cx_static_check,
+      assert_message
+        IMPORTING
+          act_message TYPE zif_aff_log=>ty_log_out
+          type        TYPE c
+          exp_message TYPE symsg.
+ENDCLASS.
+
+
+CLASS ltcl_log_unit_test IMPLEMENTATION.
+
+  METHOD setup.
+    log = NEW zcl_aff_log( ).
+    message = VALUE #( msgid = 'Z_AFF_TOOLS' msgno = 100 msgv1 = 'TEST' ).
+    message = VALUE #( msgid = 'Z_AFF_TOOLS' msgno = 1 msgv1 = 'TEST' ).
+  ENDMETHOD.
+
+
+  METHOD add_info.
+    log->add_info( message ).
+
+    DATA(messages) = log->get_messages( ).
+    cl_abap_unit_assert=>assert_equals( exp = 1 act = lines( messages ) ).
+    assert_message( act_message = messages[ 1 ] type = 'I' exp_message = message ).
+  ENDMETHOD.
+
+
+  METHOD add_warning.
+    log->add_warning( message ).
+
+    DATA(messages) = log->get_messages( ).
+    cl_abap_unit_assert=>assert_equals( exp = 1 act = lines( messages ) ).
+    assert_message( act_message = messages[ 1 ] type = 'W' exp_message = message ).
+  ENDMETHOD.
+
+
+  METHOD add_error.
+    log->add_error( message ).
+
+    DATA(messages) = log->get_messages( ).
+    cl_abap_unit_assert=>assert_equals( exp = 1 act = lines( messages ) ).
+    assert_message( act_message = messages[ 1 ] type = 'E' exp_message = message ).
+  ENDMETHOD.
+
+
+  METHOD add_t100_exception.
+    DATA(previous) = NEW zcx_aff_tools( textid = VALUE #( msgid = 'Z_AFF_TOOLS' msgno = 101 ) ).
+    DATA(exception) = NEW zcx_aff_tools( textid = VALUE #( msgid = 'Z_AFF_TOOLS' msgno = 100 ) previous = previous ).
+
+    log->add_exception( exception ).
+
+    DATA(messages) = log->get_messages( ).
+    cl_abap_unit_assert=>assert_equals( exp = 2 act = lines( messages ) ).
+    assert_message( act_message = messages[ 1 ] type = 'E' exp_message = VALUE #( msgid = 'Z_AFF_TOOLS' msgno = 100 ) ).
+    assert_message( act_message = messages[ 2 ] type = 'E' exp_message = VALUE #( msgid = 'Z_AFF_TOOLS' msgno = 101 ) ).
+  ENDMETHOD.
+
+
+  METHOD add_catched_exception.
+    TRY.
+        RAISE EXCEPTION TYPE zcx_aff_tools MESSAGE e100(z_aff_tools) WITH '1' '2' '3' '4'.
+      CATCH zcx_aff_tools INTO DATA(exception).
+        log->add_exception( exception ).
+    ENDTRY.
+
+    DATA(messages) = log->get_messages( ).
+    cl_abap_unit_assert=>assert_equals( exp = 1 act = lines( messages ) ).
+    assert_message( act_message = messages[ 1 ]
+                    type        = 'E'
+                    exp_message = VALUE #( msgid = 'Z_AFF_TOOLS' msgno = 100 msgv1 = '1' msgv2 = '2' msgv3 = '3' msgv4 = '4' ) ).
+  ENDMETHOD.
+
+
+  METHOD add_classic_exception.
+    DATA(exception) = NEW ltcx_classic_exception( ).
```

Comment:
Let's replace this by an actual exception like `zcx_aff_tools`

Reply:
Done